### PR TITLE
PLT-2940 Added validation for command triggers

### PR DESCRIPTION
**Pull request by DavidLu1997**
Command triggers containing a / can no longer be created. This prevents commands such as `//foo` causing server errors.

Future Todos:
- Validate commands server side (IsValid is not called on commands)
- Enforce restrictions on commands (e.g. letters and numbers only)

### CODE REVIEW

**Comment by crspeller** [comment]
@DavidLu1997 This fails to build.

**Comment by crspeller** [comment]
@DavidLu1997 Strange build failure. Does this work for you locally? What version of go are you running? Or did you just submit without ever running it?

**Comment by DavidLu1997** [comment]
This should now build, it ran locally for me on Go 1.6

**Comment by DavidLu1997** [comment]
This is odd
 ```
1 failing

  1) SpinnerButton "before all" hook:
     Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
      at Object.done (node_modules/mocha-jsdom/index.js:70:7)
      at node_modules/jsdom/lib/jsdom.js:296:18
      at _combinedTickCallback (internal/process/next_tick.js:67:7)
      at process._tickCallback (internal/process/next_tick.js:98:9)

**Comment by crspeller** [comment]
Ya I think we need to increase that timeout. When the server is running parallel builds that timeout sometimes fires. 

I don't think removing the server validation is the correct fix here. It should probably be included in this commit. Is there a ticket for your other TODO?

**Comment by DavidLu1997** [comment]
The server side validation is never called, so it's not useful for this PR. I created a ticket for my other TODOs

**Comment by crspeller** [comment]
I don't understand. Why don't we call it? If it's not being called why don't we call it and add the appropriate validation?

**Comment by DavidLu1997** [comment]
Wait, we do call IsValid, I'll add the changes back in

File: model/command.go [hunk 1]
```diff
@@ -99,7 +100,7 @@ func (o *Command) IsValid() *AppError {
 		return NewLocAppError("Command.IsValid", "model.command.is_valid.team_id.app_error", nil, "")
 	}
 
-	if len(o.Trigger) == 0 || len(o.Trigger) > 128 {
+	if len(o.Trigger) == 0 || len(o.Trigger) > 128 || strings.Contains(o.Trigger, "/") {
```
Comment (crspeller) [inline model/command.go]: Shouldn't this be checking if a `/` is at position 0, not just anywhere?

**Comment by jwilander** [comment]
David we missed this one, let's chat later about it

**Pull request closed without merging**